coap/api/transport.go

```diff
@@ -136,7 +145,7 @@ func decodeMessage(msg *mux.Message) (messaging.Message, error) {
 	}
 	ret := messaging.Message{
 		Protocol: protocol,
-		Channel:  parseID(path),
+		Channel:  channelParts[1],
```

Comment:
It's not rlated to your modifications but there is a length check for `channelParts  < 2` but it should be `< 3` right?

Reply:
That's correct, good catch.